lib/factorial.rb

```diff
@@ -1,4 +1,19 @@
 # Computes factorial of the input number and returns it
+require 'pry'
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError if number.nil?
+
+  product = 1
+
+  while number > 0
+    product = product * number
+    number = number - 1
+  end
+
+  return product
 end
+
+# Time complexity: linear
+# Given a number of value n, the time complexity is O(n). The number of loops executed is determined by the value of the number.
+# Space complexity: constant, O(1)?
+# Does the size of the number being stored matter for space complexity? (i.e., Is space complexity the same if factorial = 5 or factorial = 5_000_000_000_000?)
```

Comment:
Correct! The reference to the value of the number will change. But the value of the number or the value of the product will not matter in computing the space complexity.